### ISSUE TITLE
Compatibility issue with Relational Pouch

### ISSUE THREAD

**Issue by kennyki** [angular-pouchdb/angular-pouchdb#37]
When using a plugin like [Relational Pouch](https://github.com/nolanlawson/relational-pouch) the additional methods are **plugged after instantiation**. In this case, all the `db.rel.*` functions are available after `db.setSchema()`. Thus the [recommended options](https://github.com/angular-pouchdb/angular-pouchdb#options) is not applicable (getting error: 'db.rel is undefined').

I have found a workaround:

```
angular.module('myapp.db', ['pouchdb'])

.factory('db', function(pouchDB, pouchDecorators) {

  var db = pouchDB('myapp', {
    adapter : 'websql'
  });

  db.setSchema([
    // my schema
  ]);

  /* basically implementing:
  POUCHDB_METHODS.rel = {
    save: 'qify',
    find: 'qify',
    del: 'qify',
    putAttachment: 'qify',
    getAttachment: 'qify',
    removeAttachment: 'qify',
    parseDocID: 'qify',
    makeDocID: 'qify',
  };
  */
  angular.forEach(['save', 'find', 'del', 'putAttachment', 'getAttachment', 'removeAttachment', 'parseDocID', 'makeDocID'], function(fn) {
    db.rel[fn] = pouchDecorators.qify(db.rel[fn]);
  });

  return db;

})

;
```

But I'd like to know if there's a better way to do it. Has anyone else faced similar issue or have better ideas?

**Comment by nolanlawson** [comment]
@tlvince It might be reasonable to just manually add method names for the [popular PouchDB plugins](http://pouchdb.com/external.html). Unless there is a way to automatically qify?

**Comment by tlvince** [comment]
@kennyki, hmm, your workaround is probably the best way to do this right now. We should be able to come up with something that makes this easier though.

@nolanlawson, I was hoping to limit angular-pouchdb's responsibility to Pouch core and using independent modules for specific cases such as these. Perhaps something like angular-pouchdb/angular-relational-pouch would work?

I can't think of a reliable automated way to handle qify, hence the explicit enumeration. (NB, we're also [handling eventEmitters](https://github.com/angular-pouchdb/angular-pouchdb/blob/d06bcefaa7a1e0120194b096b0e3fe26dcbe5681/angular-pouchdb.js#L20)). It would be nice, though!

**Comment by nolanlawson** [comment]
`angular-relational-pouch` sounds kind of excessive. There are dozens of plugins already; no need to double their number. :stuck_out_tongue: 

I am 99% sure there is a way to wrap PouchDB's API with the qified/rootScope.apply-ified API without having to explicitly enumerate all the existing PouchDB functions. I mean, `bluebird.promisifyAll` is able to do something similar, right?

**Comment by tlvince** [comment]
> `angular-relational-pouch` sounds kind of excessive. There are dozens
> of plugins already; no need to double their number. :stuck_out_tongue:

True, but I was thinking only of `relational-pouch`; it's the only one
I've come across so far that does its stuff after instantiation (not
saying there aren't others). For the rest, the `pouchDBProvider.methods`
config should suffice.

> I am 99% sure there is a way to wrap PouchDB's API with the
> qified/rootScope.apply-ified API without having to explicitly
> enumerate all the existing PouchDB functions.

:+1: Would love to see this.

**Comment by nolanlawson** [comment]
@marten-de-vries can't pouchdb wrapper do this?

**Comment by marten-de-vries** [comment]
@nolanlawson pouchdb-wrappers only offers an advantage if you want to 1) return a promise and not care about callback and 2) don't want to manually parse arguments. So, I think it wouldn't really help here.

**Comment by FlipOneDev** [comment]
Is angular-relational-pouch totally out of the picture now?

**Comment by angelxmoreno** [comment]
@FlipOneDev i agree with @nolanlawson that making an angular plugin for an angular module to use pouchdb plugins is a bit much. I like the solution on adding the pouchdb plugin integration part to the angular-pouchdb module but @tlvince prefers the the module to be lean and only be responsible for pouchdb's core functions.

At least that is what I am interpreting from the convo here. 

I would say, for the meantime, perhaps we can use the wiki to show examples of the integration for the various pouchdb plugins. This is a popular  module and I am sure a reference page to how to integrate the various pouchdb plugins would be very helpful. Is that a good compromise @tlvince ?

**Comment by tlvince** [comment]
I'm for that, though would prefer the docs go into the readme or in a `docs` directory in the repo so they have a better chance of being maintained. Want to give it a shot? Note there's already the [Does this work with PouchDB plugins?](https://github.com/angular-pouchdb/angular-pouchdb#does-this-work-with-pouchdb-plugins) section.

**Comment by angelxmoreno** [comment]
Of course I can give it a shot! Just keep in mind that in a perfect world, we would document most, if not, all the pouch plugins on http://pouchdb.com/external.html That list could get big.

**Comment by miguelbecerra01** [comment]
Some has an example of how to implement angular-pouchdb with Relational Pouch?

**Comment by angelxmoreno** [comment]
@miguelbecerra01 did you try @kennyki 's solution?
